src/adapter/aws-lambda/handler.ts

```diff
@@ -24,41 +40,94 @@ interface APIGatewayProxyResult {
   isBase64Encoded: boolean
 }
 
+/**
+ * Accepts events from API Gateway/ELB(`APIGatewayProxyEvent`) and directly through Function Url(`APIGatewayProxyEventV2`)
+ */
 export const handle = (app: Hono) => {
-  return async (event: APIGatewayEvent): Promise<APIGatewayProxyResult> => {
+  return async (
+    event: APIGatewayProxyEvent | APIGatewayProxyEventV2
+  ): Promise<APIGatewayProxyResult> => {
     const req = createRequest(event)
     const res = await app.fetch(req)
-    const arrayBuffer = await res.arrayBuffer()
-    const result: APIGatewayProxyResult = {
-      statusCode: res.status,
-      body: String.fromCharCode(...new Uint8Array(arrayBuffer)),
-      headers: {},
-      isBase64Encoded: false,
-    }
-
-    res.headers.forEach((value, key) => {
-      result.headers[key] = value
-    })
-
-    return result
+
+    return createResult(res)
+  }
+}
+
+const createResult = async (res: Response): Promise<APIGatewayProxyResult> => {
+  const isBase64Encoded = res.body instanceof ReadableStream
```

Comment:
I'm not sure about this, and should look into it, but `res.body` will always be a `ReadableStream`? What do you think?